### ISSUE TITLE
Replace jQuery Tools

### ISSUE THREAD

**Issue by Rumperuu** [markcheret/footnotes#85]
jQuery Tools is an ancient, dead library that should be replaced as a matter of urgency.

See discussion at https://github.com/markcheret/footnotes/pull/72

**Comment by lolzim** [comment]
Since WP 5.6, jQuery 3.5.1 is in wp-includes/js/jquery/
Up to 5.5, version 1.12.4 was embeded (!)

So, stats should be that there is still a large base of 1.12.4. Hence we probably cannot rely on 3.5.1 for choosing a replacement.

**Comment by markcheret** [comment]
is it a solution to make the Plugin only WP >5.6 compatible?

**Comment by lolzim** [comment]
@markcheret that would at least mean fn future versions are only for 46% of installed WP, according to WP's stats:
https://wordpress.org/about/stats/

**Comment by markcheret** [comment]
I don't know which of our users still use an ancient version of WordPress and at some point supporting an old version of WP won't do us any good. Since this is also security-related I highly recommend 5.6 as a minimum requirement to also encourage users to update.

**Comment by Rumperuu** [comment]
[jQuery UI](https://jqueryui.com/) would be the most obvious candidate for a replacement (I'm assuming it has tooltips), and it looks like it supports jQuery 1.7+ so that should be fine.

That said, given the advances in what can be done with pure CSS since the Plugin started, we might not even need a JavaScript solution at all (see #44)

**Comment by markcheret** [comment]
OK. So what's the low-hanging fruit here with maximum security and usability impact and lowest possible development effort?

**Comment by lolzim** [comment]
jQUery UI seems to be abandonware since 2016:
https://jqueryui.com/download/all/

**Comment by Rumperuu** [comment]
> jQUery UI seems to be abandonware since 2016:
> https://jqueryui.com/download/all/

Good spot — searching around it looks like it's _officially_ still alive, but functionally as dead as disco. Although the general consensus seems to be that this is because it's no longer necessary due to advances in JavaScript (and, presumably, CSS) capability.

@markcheret I think the risk posed by jQuery Tools, no matter how outdated it may be, is likely to be minimal; it's a tiny UI library, so it's unlikely to present much of an attack surface. I think the best bet would be to review what we're actually using it for (i.e. as part of #87) and then determine how best to replace it, as if it's just being used for showing tooltips then we'll probably be able to lose the dependency entirely, but if it's being used for a bunch of other stuff then it may be worth looking into replacement libraries.

**Comment by markcheret** [comment]
OK. Great.
Then this can be moved to discussions? Or even closed?

**Comment by Rumperuu** [comment]
> OK. Great.
> Then this can be moved to discussions? Or even closed?

I think leave this Issue open for now, and we can come back to it once #87 is done.

**Comment by markcheret** [comment]
fair enough :)
